bitbot/markov.go

```diff
@@ -38,6 +39,35 @@ var MarkovResponseTrigger = NamedTrigger{
 	},
 }
 
+var MarkovInitTrigger = NamedTrigger{
+	ID:   "markovInit",
+	Help: "Resets markov chain to a fresh chain, or bootstraps it with sample texts. Usage: !markov reset, !markov init",
+	Condition: func(irc *hbot.Bot, m *hbot.Message) bool {
+		return m.Command == "PRIVMSG" && strings.HasPrefix(m.Content, "!markov ")
+	},
+	Action: func(irc *hbot.Bot, m *hbot.Message) bool {
+		cmd := strings.Split(m.Content, " ")
+		if len(cmd) < 2 {
+			return false
+		}
+		switch cmd[1] {
+		case "reset":
+			b.mChain = gomarkov.NewChain(1)
+			return true
+		case "init":
+			b.mChain = gomarkov.NewChain(1)
+			if markovInit(b.mChain) == false {
```

Comment:
You can actually flip this around. You don't need to check equivalence with a bool here, since you're just getting a bool back again.

```
if markovInit(b.mChain) {
    // true condition
} else {
    // false condition
}
```